src/overlays/actors/ovl_En_Sob1/z_en_sob1.c

```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
+                return 0xBBD;
+            }
+            gSaveContext.weekEventReg[58] |= 0x10;
+            return 0xBBC;
+        } else {
+            if (gSaveContext.weekEventReg[58] & 0x20) {
+                return 0xBBF;
+            }
+            gSaveContext.weekEventReg[58] |= 0x20;
+            return 0xBBE;
+        }
+    }
+    return 0;
+}
+
+u16 EnSob1_GetGoodbye(EnSob1* this) {
+    if (this->shopType == BOMB_SHOP) {
+        if (gSaveContext.day == 1) {
+            return 0x64C;
+        } else if (gSaveContext.day == 2) {
+            return 0x64D;
+        } else if (!gSaveContext.isNight) {
+            return 0x64E;
+        } else {
+            return 0x64F;
+        }
+    }
+    return 0x64C;
+}
+
+void EnSob1_EndInteractionBombShop(EnSob1* this, GlobalContext* globalCtx) {
+    this->drawCursor = 0;
+    this->stickLeftPrompt.isEnabled = false;
+    this->stickRightPrompt.isEnabled = false;
+    this->goodbyeTextId = EnSob1_GetGoodbye(this);
+    func_801518B0(globalCtx, this->goodbyeTextId, &this->actor);
+    EnSob1_SetupAction(this, EnSob1_EndingInteraction);
+}
+
+void EnSob1_SpawnShopItems(EnSob1* this, GlobalContext* globalCtx, ShopItem* shopItem) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->items); i++, shopItem++) {
+        if (shopItem->shopItemId < 0) {
+            this->items[i] = NULL;
+        } else {
+            this->items[i] =
+                (EnGirlA*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_GIRLA, shopItem->spawnPos.x,
+                                      shopItem->spawnPos.y, shopItem->spawnPos.z, 0, 0, 0, shopItem->shopItemId);
+        }
+    }
+}
+
+s32 EnSob1_GetObjIndicies(EnSob1* this, GlobalContext* globalCtx, s16* objIds) {
+    if (objIds[1] != OBJECT_ID_MAX) {
+        this->objIndicies[1] = Object_GetIndex(&globalCtx->objectCtx, objIds[1]);
+        if (this->objIndicies[1] < 0) {
+            return false;
+        }
+    } else {
+        this->objIndicies[1] = -1;
+    }
+    if (objIds[2] != OBJECT_ID_MAX) {
+        this->objIndicies[2] = Object_GetIndex(&globalCtx->objectCtx, objIds[2]);
+        if (this->objIndicies[2] < 0) {
+            return false;
+        }
+    } else {
+        this->objIndicies[2] = -1;
+    }
+    return true;
+}
+
+void EnSob1_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSob1* this = THIS;
+    s32 pad;
+    s16* objIds;
+
+    switch (ENSOB1_GET_SHOPTYPE(thisx)) {
+        case ZORA_SHOP:
+            this->shopType = ZORA_SHOP;
+            break;
+        case GORON_SHOP:
+            if (gSaveContext.weekEventReg[33] & 0x80) {
+                this->shopType = GORON_SHOP_SPRING;
+            } else {
+                this->shopType = GORON_SHOP;
+            }
+            break;
+        case BOMB_SHOP:
+            this->shopType = BOMB_SHOP;
+            break;
+        default:
+            Actor_MarkForDeath(&this->actor);
+            return;
+    }
+
+    objIds = sObjectIds[this->shopType];
```

Comment:
I think the uses of this variable would be easier to follow if there was at least an enum for the entry indices (something like what the three objects are actually used for).

Reply:
That is a good idea, though I don't know enough about the objects yet to that. I've added a todo to add that enum when the objects are better understood.